File: js/283_MoveZeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
	var j = 0;
	for( var i in nums ){
		if( nums[i] !== 0 ){
			var tmp = nums[i];
			nums[i] = nums[j];
			nums[j] = tmp;
			++j;
		}
	}
};

File: cpp/283_MoveZeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int j = 0;
        for(size_t i = 0; i < nums.size(); ++i){
            if( nums[i] != 0 ){
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                ++j;
            }
        }
    }
};